Bottom: 8e17ce44d915a61a37514e7d8adbefa73fe03378
Top:    ab17e4d95123751782ce5365e4a9b290e0ce7050
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-10-19 17:47:55 +0100

hid/common: Fix dereference bug in hid_find_action().

We want to check context for being NULL, not *context.


---

diff --git a/src/hid/common/actions.c b/src/hid/common/actions.c
index 4bd886b..8678ad9 100644
--- a/src/hid/common/actions.c
+++ b/src/hid/common/actions.c
@@ -180,7 +180,7 @@ hid_find_action (const char *name, void **context)
 
   for (i = 0; i < n_actions; i++)
     if (strcasecmp (all_actions[i].action.name, name) == 0) {
-      if (*context != NULL)
+      if (context != NULL)
         *context = all_actions[i].context;
       return &(all_actions[i].action);
     }
